qoqo/src/devices/all_to_all.rs

```diff
@@ -48,7 +48,7 @@ impl AllToAllDeviceWrapper {
     ///     default_gate_time (float): The default starting gate time.
     ///
     /// Returns:
-    ///     AllToAllDevice
+    ///     Self: The new device with the new properties
```

Comment:
Could you apply something like this to the SquareLatticeDevice too? It's indeed a bit more clear.